From 3cd3c717bae02a8668a8d5303bef875a484b874d Mon Sep 17 00:00:00 2001
From: "A. Maitland Bottoms" <bottoms@debian.org>
Date: Fri, 26 Oct 2018 21:10:34 -0400
Subject: [PATCH 1/6] use SOURCE_DATE_EPOCH

Reproducible builds set SOURCE_DATE_EPOCH rather than arbitrary
build timestamps. Use it if set, otherwise revert to the old way.
---
 host/cmake/Modules/UHDBuildInfo.cmake | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/host/cmake/Modules/UHDBuildInfo.cmake b/host/cmake/Modules/UHDBuildInfo.cmake
index 21b1e3c1d..e7bd282c3 100644
--- a/host/cmake/Modules/UHDBuildInfo.cmake
+++ b/host/cmake/Modules/UHDBuildInfo.cmake
@@ -14,7 +14,12 @@ MACRO(UHD_LOAD_BUILD_INFO)
 
     # Build date
     IF(IGNORE_BUILD_DATE)
-        SET(UHD_BUILD_DATE "")
+      SET(UHD_BUILD_DATE "")
+    ELSEIF(DEFINED ENV{SOURCE_DATE_EPOCH})
+        EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c
+            "import time; print(time.strftime('%a, %d %b %Y %H:%M:%S UTC', time.gmtime($ENV{SOURCE_DATE_EPOCH})))"
+            OUTPUT_VARIABLE UHD_BUILD_DATE OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
     ELSE()
         EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c
             "import time; print(time.strftime('%a, %d %b %Y %H:%M:%S', time.gmtime()))"
-- 
2.11.0

